data_structures/linked_list/linked_list.py

```diff
@@ -2,35 +2,57 @@
 
 
 class LinkedList:
+    """Linked list data structure"""
     def __init__(self, iter=[]):
+        """Initialize linked list of size length."""
         self.head = None
         self._size = 0
 
         try:
             for item in reversed(iter):
                 self.insert(item)
         except TypeError:
-            print('Not an iterable')
+            raise TypeError('Please pass an iterable.')
```

Comment:
So take all of this out of the try, except and wrap the raise in an if before the for loop. There is no case where your insert will actually raise a type error.